src/DynamoCoreWpf/ViewModels/Core/WorkspaceViewModel.cs

```diff
@@ -1054,12 +1054,31 @@ private void DoGraphAutoLayout(object o)
         {
             if (Model.Nodes.Count() == 0)
                 return;
+            
+            var selection = DynamoSelection.Instance.Selection;
+
+            bool isInternalLayout = selection.Count > 0 &&
+                selection.All(x => x is AnnotationModel ||
+                selection.Where(g => g is AnnotationModel)
+                .Any(g => (g as AnnotationModel).SelectedModels.Contains(x)));
```

Comment:
Wow! Too much stuff being done in one line of code. Perhaps a comment here will help developers reading this.